Library crashes when no MAC address specified in ScanFilter

**Describe the bug**
Application crash when try scan devices when ScanFilter has `null` device mac address

**To Reproduce**
Steps to reproduce the behavior:
1. Create a `ScanFilter` and don't specify device mac address
2. Perform `RxBleClient.scanBleDevices(...)`
3. `NullPointerException` occurs

**Expected behavior**
Application start scanning bluetooth devices

**Smartphone (please complete the following information):**
 - Device: Google Pixel 4XL
 - OS: Android 12 Developer Preview 3
 - Library version: 1.12.1
 
**Exception stacktrace**

```
java.lang.NullPointerException
    at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:147)
    at android.bluetooth.le.ScanFilter$Builder.setDeviceAddressInternal(ScanFilter.java:678)
    at android.bluetooth.le.ScanFilter$Builder.setDeviceAddress(ScanFilter.java:615)
    at android.bluetooth.le.ScanFilter$Builder.setDeviceAddress(ScanFilter.java:590)
    at com.polidea.rxandroidble2.internal.scan.AndroidScanObjectsConverter.toNative(AndroidScanObjectsConverter.java:71)
    at com.polidea.rxandroidble2.internal.scan.AndroidScanObjectsConverter.toNativeFilters(AndroidScanObjectsConverter.java:56)
    at com.polidea.rxandroidble2.internal.operations.ScanOperationApi21.startScan(ScanOperationApi21.java:117)
    at com.polidea.rxandroidble2.internal.operations.ScanOperationApi21.startScan(ScanOperationApi21.java:28)
    at com.polidea.rxandroidble2.internal.operations.ScanOperation.protectedRun(ScanOperation.java:43)
    at com.polidea.rxandroidble2.internal.QueueOperation$1.subscribe(QueueOperation.java:41)
    at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
    at io.reactivex.Observable.subscribe(Observable.java:12284)
    at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn.subscribeActual(ObservableUnsubscribeOn.java:32)
    at io.reactivex.Observable.subscribe(Observable.java:12284)
    at com.polidea.rxandroidble2.internal.serialization.FIFORunnableEntry$1.run(FIFORunnableEntry.java:56)
    at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
    at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    at java.lang.Thread.run(Thread.java:920)
```

**Additional context**
The issue is related to OS only and should not be realted to device of library version (at least few last ones)
The problem here is inside `AndroidScanObjectsConverter.toNative(scanFilter)`
As the `Device Address` inside `ScanFilter` is `null`, the sdk's `ScanFilter.Builder` verifies that the value is not null and throws exception. 
`toNative()` method should set only `not null` values to the builder. Seems that `null` chedking was added in Android 12 DP 3

👋 hello -

I pulled down the `fix/android_12_scan_filter_null_mac_address` branch to try it in our app and I am receiving the following exception on my Android 12 device. Please let me know if there is any additional info I can help provide!

```    java.lang.NoClassDefFoundError: Failed resolution of: Lcom/jakewharton/rxrelay2/BehaviorRelay;
        at com.polidea.rxandroidble2.internal.DeviceModule.provideConnectionStateRelay(DeviceModule.java:57)
        at com.polidea.rxandroidble2.internal.DeviceModule_ProvideConnectionStateRelayFactory.get(DeviceModule_ProvideConnectionStateRelayFactory.java:17)
        at com.polidea.rxandroidble2.internal.DeviceModule_ProvideConnectionStateRelayFactory.get(DeviceModule_ProvideConnectionStateRelayFactory.java:9)
        at bleshadow.dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at com.polidea.rxandroidble2.internal.RxBleDeviceImpl_Factory.get(RxBleDeviceImpl_Factory.java:31)
        at com.polidea.rxandroidble2.internal.RxBleDeviceImpl_Factory.get(RxBleDeviceImpl_Factory.java:11)
        at bleshadow.dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
        at com.polidea.rxandroidble2.DaggerClientComponent$DeviceComponentImpl.provideDevice(DaggerClientComponent.java:509)
        at com.polidea.rxandroidble2.internal.RxBleDeviceProvider.getBleDevice(RxBleDeviceProvider.java:43)
        at com.polidea.rxandroidble2.internal.scan.InternalToExternalScanResultConverter.apply(InternalToExternalScanResultConverter.java:26)
        at com.polidea.rxandroidble2.internal.scan.InternalToExternalScanResultConverter.apply(InternalToExternalScanResultConverter.java:13)
        at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:57)
        at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn$UnsubscribeObserver.onNext(ObservableUnsubscribeOn.java:60)
        at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:66)
        at com.polidea.rxandroidble2.internal.serialization.FIFORunnableEntry$1$1.onNext(FIFORunnableEntry.java:68)
        at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn$UnsubscribeObserver.onNext(ObservableUnsubscribeOn.java:60)
        at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:66)
        at com.polidea.rxandroidble2.internal.operations.ScanOperationApi21$1.onScanResult(ScanOperationApi21.java:83)
        at android.bluetooth.le.BluetoothLeScanner$BleScanCallbackWrapper$1.run(BluetoothLeScanner.java:495)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7727)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:975)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.jakewharton.rxrelay2.BehaviorRelay" on path: DexPathList[[zip file "/data/app/~~L_Ht2APB6tyHIvWNSunmPQ==/PACKAGENAME-zgPpgInIuWiwKZ8rlNeMng==/base.apk"],nativeLibraryDirectories=[/data/app/~~L_Ht2APB6tyHIvWNSunmPQ==/PACKAGENAME-zgPpgInIuWiwKZ8rlNeMng==/lib/arm64, /data/app/~~L_Ht2APB6tyHIvWNSunmPQ==/PACKAGENAME-zgPpgInIuWiwKZ8rlNeMng==/base.apk!/lib/arm64-v8a, /system/lib64, /system_ext/lib64, /product/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:207)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.polidea.rxandroidble2.internal.DeviceModule.provideConnectionStateRelay(DeviceModule.java:57) 
        at com.polidea.rxandroidble2.internal.DeviceModule_ProvideConnectionStateRelayFactory.get(DeviceModule_ProvideConnectionStateRelayFactory.java:17) 
        at com.polidea.rxandroidble2.internal.DeviceModule_ProvideConnectionStateRelayFactory.get(DeviceModule_ProvideConnectionStateRelayFactory.java:9) 
        at bleshadow.dagger.internal.DoubleCheck.get(DoubleCheck.java:47) 
        at com.polidea.rxandroidble2.internal.RxBleDeviceImpl_Factory.get(RxBleDeviceImpl_Factory.java:31) 
        at com.polidea.rxandroidble2.internal.RxBleDeviceImpl_Factory.get(RxBleDeviceImpl_Factory.java:11) 
        at bleshadow.dagger.internal.DoubleCheck.get(DoubleCheck.java:47) 
        at com.polidea.rxandroidble2.DaggerClientComponent$DeviceComponentImpl.provideDevice(DaggerClientComponent.java:509) 
        at com.polidea.rxandroidble2.internal.RxBleDeviceProvider.getBleDevice(RxBleDeviceProvider.java:43) 
        at com.polidea.rxandroidble2.internal.scan.InternalToExternalScanResultConverter.apply(InternalToExternalScanResultConverter.java:26) 
        at com.polidea.rxandroidble2.internal.scan.InternalToExternalScanResultConverter.apply(InternalToExternalScanResultConverter.java:13) 
        at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:57) 
        at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn$UnsubscribeObserver.onNext(ObservableUnsubscribeOn.java:60) 
        at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:66) 
        at com.polidea.rxandroidble2.internal.serialization.FIFORunnableEntry$1$1.onNext(FIFORunnableEntry.java:68) 
        at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn$UnsubscribeObserver.onNext(ObservableUnsubscribeOn.java:60) 
        at io.reactivex.internal.operators.observable.ObservableCreate$CreateEmitter.onNext(ObservableCreate.java:66) 
        at com.polidea.rxandroidble2.internal.operations.ScanOperationApi21$1.onScanResult(ScanOperationApi21.java:83) 
        at android.bluetooth.le.BluetoothLeScanner$BleScanCallbackWrapper$1.run(BluetoothLeScanner.java:495) 
        at android.os.Handler.handleCallback(Handler.java:938) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7727) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:975) 
```

I am currently testing on the beta and am still seeing that same exception in the original post

```    com.polidea.rxandroidble2.exceptions.BleScanException: Bluetooth cannot start (code 0)
        at com.polidea.rxandroidble2.internal.operations.ScanOperation.protectedRun(ScanOperation.java:50)
        at com.polidea.rxandroidble2.internal.QueueOperation$1.subscribe(QueueOperation.java:41)
        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40)
        at io.reactivex.Observable.subscribe(Observable.java:12284)
        at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn.subscribeActual(ObservableUnsubscribeOn.java:32)
        at io.reactivex.Observable.subscribe(Observable.java:12284)
        at com.polidea.rxandroidble2.internal.serialization.FIFORunnableEntry$1.run(FIFORunnableEntry.java:56)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:920)
     Caused by: java.lang.NullPointerException
        at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:147)
        at android.bluetooth.le.ScanFilter$Builder.setDeviceAddressInternal(ScanFilter.java:678)
        at android.bluetooth.le.ScanFilter$Builder.setDeviceAddress(ScanFilter.java:615)
        at android.bluetooth.le.ScanFilter$Builder.setDeviceAddress(ScanFilter.java:590)
        at com.polidea.rxandroidble2.internal.scan.AndroidScanObjectsConverter.toNative(AndroidScanObjectsConverter.java:71)
        at com.polidea.rxandroidble2.internal.scan.AndroidScanObjectsConverter.toNativeFilters(AndroidScanObjectsConverter.java:56)
        at com.polidea.rxandroidble2.internal.operations.ScanOperationApi21.startScan(ScanOperationApi21.java:104)
        at com.polidea.rxandroidble2.internal.operations.ScanOperationApi21.startScan(ScanOperationApi21.java:28)
        at com.polidea.rxandroidble2.internal.operations.ScanOperation.protectedRun(ScanOperation.java:43)
        at com.polidea.rxandroidble2.internal.QueueOperation$1.subscribe(QueueOperation.java:41) 
        at io.reactivex.internal.operators.observable.ObservableCreate.subscribeActual(ObservableCreate.java:40) 
        at io.reactivex.Observable.subscribe(Observable.java:12284) 
        at io.reactivex.internal.operators.observable.ObservableUnsubscribeOn.subscribeActual(ObservableUnsubscribeOn.java:32) 
        at io.reactivex.Observable.subscribe(Observable.java:12284) 
        at com.polidea.rxandroidble2.internal.serialization.FIFORunnableEntry$1.run(FIFORunnableEntry.java:56) 
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) 
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:920) 
```

--- a/rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/scan/AndroidScanObjectsConverter.java
+++ b/rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/scan/AndroidScanObjectsConverter.java
@@ -67,8 +67,10 @@ public class AndroidScanObjectsConverter {
         if (scanFilter.getServiceDataUuid() != null) {
             builder.setServiceData(scanFilter.getServiceDataUuid(), scanFilter.getServiceData(), scanFilter.getServiceDataMask());
         }
+        if (scanFilter.getDeviceAddress() != null) {
+            builder.setDeviceAddress(scanFilter.getDeviceAddress());
+        }
         return builder
-                .setDeviceAddress(scanFilter.getDeviceAddress())
                 .setDeviceName(scanFilter.getDeviceName())
                 .setManufacturerData(scanFilter.getManufacturerId(), scanFilter.getManufacturerData(), scanFilter.getManufacturerDataMask())
                 .setServiceUuid(scanFilter.getServiceUuid(), scanFilter.getServiceUuidMask())

